static/app/views/replays/detail/header/replayDetailsUserBadge.tsx

```diff
@@ -20,6 +21,26 @@ interface Props {
 export default function ReplayDetailsUserBadge({readerResult}: Props) {
   const replayRecord = readerResult.replayRecord;
   const replay = readerResult.replay;
+
+  const tickerRef = useRef<number | undefined>(undefined);
+  const [isLive, setIsLive] = useState(replay?.getIsLive());
+
+  useEffect(() => {
+    // If the replay is live, start the ticker
+    if (replay?.getIsLive()) {
+      const ONE_MINUTE_INTERVAL = 60 * 1000;
+      tickerRef.current = window.setInterval(() => {
+        setIsLive(replay?.getIsLive());
+
+        if (!replay?.getIsLive()) {
+          window.clearInterval(tickerRef.current);
+        }
+      }, ONE_MINUTE_INTERVAL);
+    }
+
+    return () => window.clearInterval(tickerRef.current);
+  }, [replay]);
```

Comment:
technically you don't need the `useRef`. It could be a local variable inside the `useEffect`.  It'll get closed-over inside that anon function and work the same

```suggestion
  const [isLive, setIsLive] = useState(replay?.getIsLive());

  useEffect(() => {
    // If the replay is live, start the ticker
    let interval: null | number = null;
    if (replay?.getIsLive()) {
      const ONE_MINUTE_INTERVAL = 60 * 1000;
      interval = window.setInterval(() => {
        setIsLive(replay?.getIsLive());

        if (!replay?.getIsLive()) {
          window.clearInterval(interval);
          interval = null;
        }
      }, ONE_MINUTE_INTERVAL);
    }

    return () => {
      if (interval) {
        window.clearInterval(interval);
      }
    };
  }, [replay]);
```